lib/puppet/provider/apt_key/apt_key.rb

```diff
@@ -136,6 +136,16 @@ def tempfile(content)
     file = Tempfile.new('apt_key')
     file.write content
     file.close
+    #we now need to confirm that the fingerprint in this file, matches the long key that the user has given in the manifest
+    if name.size == 40
+      if File.executable? "/usr/bin/gpg"
```

Comment:
gpg should be there, i dont think we should fail. A warning to suggest that we cannot verify that the ID fingerprint and the source/content fingerprint are the same.